apps/platform/src/app/(main)/project/[project]/@secret/page.tsx

```diff
@@ -28,94 +28,102 @@ import {
   TooltipProvider,
   TooltipTrigger
 } from '@/components/ui/tooltip'
+import Loading from '@/components/ui/loading' // Import Loading component
 
 extend(relativeTime)
 
 function SecretPage(): React.JSX.Element {
   const [allSecrets, setAllSecrets] = useState<Secret[]>()
   const pathname = usePathname()
+  const [loading, setLoading] = useState(true); // State to manage loading state
 
   useEffect(() => {
     Secrets.getAllSecretbyProjectId(pathname.split('/')[2])
       .then((data) => {
         setAllSecrets(data)
+        setLoading(false); //  loading becomes false once data is fetched
       })
       .catch((error) => {
         // eslint-disable-next-line no-console -- we need to log the error
         console.error(error)
+        setLoading(false); // Handling loading state in case of error
       })
   }, [pathname])
 
   return (
-    <ScrollArea className=" mb-4 h-[50rem]">
-      <Accordion
-        className="flex h-[50rem] flex-col gap-4"
-        collapsible
-        type="single"
-      >
-        {allSecrets?.map((secret) => {
-          return (
-            <AccordionItem
-              className="rounded-xl bg-white/5 px-5"
-              key={secret.secret.id}
-              value={secret.secret.id}
-            >
-              <AccordionTrigger
-                className="hover:no-underline"
-                rightChildren={
-                  <div className="text-xs text-white/50">
-                    {dayjs(secret.secret.updatedAt).toNow(true)} ago by{' '}
-                    <span className="text-white">
-                      {secret.secret.lastUpdatedBy.name}
-                    </span>
-                  </div>
-                }
+    <ScrollArea className="mb-4 h-[50rem]">
+      {loading ? (
+        // Rendering Loading when data is loading
+        <Loading />
+      ) : (
```

Comment:
Try to implement react Suspense to render the loading state
refer to 
- https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming
- https://react.dev/reference/react/Suspense